crates/telio-wg/src/adapter/boring.rs

```diff
@@ -81,4 +86,49 @@ impl Adapter for BoringTun {
         self.device.read().await.trigger_exit();
         self.device.write().await.wait();
     }
+
+    async fn inject_reset_packets(&self, exit: &PublicKey) {
+        let Some(cb) = self.reset_conns_cb.as_ref() else {
+            return;
+        };
+
+        telio_log_debug!("Reseting existing connection with packet injection");
+
+        struct Sink4<'a> {
+            tun: &'a TunSocket,
+        }
+
+        struct Sink6<'a> {
+            tun: &'a TunSocket,
+        }
+
+        impl<'a> io::Write for Sink4<'a> {
+            fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+                Ok(self.tun.write4(buf))
+            }
+
+            fn flush(&mut self) -> io::Result<()> {
+                Ok(())
+            }
+        }
+
+        impl<'a> io::Write for Sink6<'a> {
+            fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+                Ok(self.tun.write6(buf))
+            }
+
+            fn flush(&mut self) -> io::Result<()> {
+                Ok(())
+            }
+        }
+
+        let dev = self.device.read().await;
+        let dev = dev.device.read();
+        let tun = dev.iface();
+
+        let mut sink4 = Sink4 { tun };
+        let mut sink6 = Sink6 { tun };
```

Comment:
Not sure, if this would change something, but maybe we should we limit IPv6 socket reset with IPv6 feature flag?

Reply:
Do you mean guarding IPv6 resets under the feature flag? Honestly, I doubt it would be useful, the IPv6 isn't implemented yet and probably never will

Reply:
The IPv6 is already implemented and ipv6-through-exit-node is still under consideration.

Reply:
I meant this feature for IPv6, not the IPv6 as a whole